src/main/java/com/google/sps/servlets/ListOrganizationsServlet.java

```diff
@@ -24,4 +41,61 @@
 @WebServlet("/list-organizations")
 public class ListOrganizationsServlet extends HttpServlet {
 
-}
\ No newline at end of file
+  /*
+   * This get request returns a list of organizations depending on its query parameters. 
+   * If no parameters are included, it will return a default list of organizations
+   */
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    /* All get requests will return a maximum of 5 organization entities */
+    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(5);
+    
+    /* TODO: Implement using pagination cursors- first implementation will just return 5 first entries */
+
+    /* This parameter is used to tell the servlet whether the user is requesting to see just the orgs they moderate*/
+    String displayUserOrgsParameter = request.getParameter("displayUserOrgs");
+    boolean displayUserOrgs = false;
+
+    /* If a parameter was sent & is set to 'true', then the displayUserOrgs boolean changes to true */
+    if ((displayUserOrgsParameter != null) && (displayUserOrgsParameter.equals("true"))) {
+      displayUserOrgs = true;
+    }
+
+    UserService userService = UserServiceFactory.getUserService();
+    boolean isUserLoggedIn = userService.isUserLoggedIn();
+
+    Query query;
+
+    if (isUserLoggedIn && displayUserOrgs) {
+      /* If the user is logged in and wants to just see their orgs, get their user ID & index with it*/
+      String userId = userService.getCurrentUser().getUserId();
+
+      query = new Query("Distributor").setFilter(new FilterPredicate("moderatorList",
+                                                                        FilterOperator.EQUAL, userId))
+                                                                        .addSort("creationTimeStamp", SortDirection.DESCENDING);
+    } else {
+      /* If no username was included, it just returns all orgs */
+        // TODO: make this only return approved orgs
+      query = new Query("Distributor").addSort("creationTimeStamp", SortDirection.DESCENDING);
```

Comment:
This behavior (logged in vs not) would be great for a test! Suggestion: Can you pull out a function "Query ConstructQueryForUserInfo(User user)" to encapsulate the idea? This makes that unit of work testable